ui/jobs/components/sam.js

```diff
@@ -1,6 +1,121 @@
 import EffectId from '../../../resources/effect_id';
+import { kAbility } from '../constants';
 
 export function setup(bars) {
-  bars.onYouGainEffect(EffectId.Shifu, () => bars.speedBuffs.shifu = 1);
-  bars.onYouLoseEffect(EffectId.Shifu, () => bars.speedBuffs.shifu = 0);
+  const comboTimer = bars.addTimerBar({
+    id: 'sam-timers-combo',
+    fgColor: 'combo-color',
+  });
+  bars.onCombo((skill) => {
+    comboTimer.duration = 0;
+    if (bars.combo.isFinalSkill)
+      return;
+    if (skill)
+      comboTimer.duration = 15;
+  });
+
+  const senContainer = document.createElement('div');
+  senContainer.id = 'sam-stacks';
+  bars.addJobBarContainer().appendChild(senContainer);
+  const sen = [
+    document.createElement('div'),
+    document.createElement('div'),
+    document.createElement('div'),
+  ];
+  sen[0].id = 'sam-stacks-setsu';
+  sen[1].id = 'sam-stacks-getsu';
+  sen[2].id = 'sam-stacks-ka';
+  sen.forEach((e) => senContainer.appendChild(e));
+
+  const kenkiGauge = bars.addResourceBox({
+    classList: ['sam-color-kenki'],
+  });
+  const meditationGauge = bars.addResourceBox({
+    classList: ['sam-color-meditation'],
+  });
+
+  bars.onJobDetailUpdate((jobDetail) => {
+    kenkiGauge.innerText = jobDetail.kenki;
+    meditationGauge.innerText = jobDetail.meditationStacks;
+    if (jobDetail.kenki >= 70)
+      kenkiGauge.parentNode.classList.add('high');
+    else
+      kenkiGauge.parentNode.classList.remove('high');
+    if (jobDetail.meditationStacks >= 2)
+      meditationGauge.parentNode.classList.add('high');
+    else
+      meditationGauge.parentNode.classList.remove('high');
+
+    if (jobDetail.setsu > 0)
+      sen[0].classList.add('active');
+    else
+      sen[0].classList.remove('active');
+    if (jobDetail.getsu > 0)
+      sen[1].classList.add('active');
+    else
+      sen[1].classList.remove('active');
+    if (jobDetail.ka > 0)
+      sen[2].classList.add('active');
+    else
+      sen[2].classList.remove('active');
```

Comment:
```suggestion
    if (jobDetail.setsu)
      sen[0].classList.add('active');
    else
      sen[0].classList.remove('active');
    if (jobDetail.getsu)
      sen[1].classList.add('active');
    else
      sen[1].classList.remove('active');
    if (jobDetail.ka)
      sen[2].classList.add('active');
    else
      sen[2].classList.remove('active');
```

setsu, getsu and ka are booleans.